owners/test/reviewer_selection.test.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2019 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const sinon = require('sinon');
+const {ReviewerSelection} = require('../src/reviewer_selection');
+const {OwnersRule, OwnersTree} = require('../src/owners');
+
+describe('reviewer selection', () => {
+  const sandbox = sinon.createSandbox();
+  const ownersTree = new OwnersTree();
+
+  const rootDirRule = new OwnersRule('OWNERS.yaml', ['root']);
+  const childDirRule = new OwnersRule('foo/OWNERS.yaml', ['child']);
+  const otherChildDirRule = new OwnersRule('biz/OWNERS.yaml', ['child', 'kid']);
+  const thirdChildDirRule = new OwnersRule('buzz/OWNERS.yaml', ['thirdChild']);
+  const descendantDirRule = new OwnersRule('foo/bar/baz/OWNERS.yaml', [
+    'descendant',
+  ]);
+
+  const rootDirTree = ownersTree.addRule(rootDirRule);
```

Comment:
Might be worth making this an object, where the keys are the directory path:
```javascript
dirTrees = {
  '/': ownersTree.addRule(...),
  '/foo': ownersTree.addRule(...),
  '/foo/bar/baz': ownersTree.addRule(...),
  ...
};
```

This way when you make comparisons below it's much more obvious without having to parse the relationships between the owner and the tree while reading the tests:
```javascript
expect(nearestTrees).toEqual([trees['/foo']])
```

Reply:
Done.